src/components/AttributeSection.vue

```diff
@@ -1,30 +1,33 @@
-<!--
-  Displays entire section of attributes
- receives events from child component and check if change is possible in resources
- if changes are possible, emit event to top component to make changes, otherwise don't
- -->
-
-<script>
+<script setup>
+/**
+ * Displays entire section of attributes
+ * receives events from child component and check if change is possible in resources
+ * if changes are possible, emit event to top component to make changes, otherwise don't
+ */
 import { statSectionMixin } from "./mixins/statSectionMixin";
 import { attributesAndSkillsMixin } from "./mixins/attributesAndSkillsMixin";
 import Stat from "./Stat.vue";
 import RestrictionState from "./RestrictionState.vue";
 
+const emit = defineEmits(['statSectionHover'])
```

Comment:
ono to nefunguje ked je to v tom kluci emits? alebo preco je to v takomto zapise
osobne sa mi moc nepaci a radsej by som pouzival dalej ten `emits:['emit1','emit2']` alebo `emits:{emit1:validator, emit2:validator}` ci co je ten ten druhy clen toho parovania

Reply:
https://v3.vuejs.org/api/sfc-script-setup.html#defineprops-and-defineemits

tu pisu, ze v SFC sa to robi takto

Reply:
https://v3.vuejs.org/api/sfc-spec.html#intro
hmm a neni to ze v script setup pre sfc sa to robi takto?

Reply:
moja otazka bola ci to funguje v kombinacii: importy v composition API a emity, props v options API

Reply:
nerozumiem otazke,... sorry :-) 
proste som to opisal ako hovoria, ze sa to v SFC moze takto pouzivat a pisu, ze to je lepsie, to je cele

Reply:
no ked som si dal tu stranku s introduction pre sfc tak som tam nic take nevidel...
len ked si das tu script setup cast tak tam to je
fuf tak snad do tretice vseho dobreho: dobre a mne by sa to viac pacilo import v setupu, props, emits v normal script, fungovalo by to?

Reply:
dal som to ako chces, aj ked mi to pride, ze potom sa ten setup sugar vobec nemusi davat...